evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -197,23 +179,16 @@ def predict(self, X, y=None):
             ValueError: If X was passed to `fit` but not passed in `predict`.
         """
         X, y = self._manage_woodwork(X, y)
-        dates, X = self._get_dates(X, y)
-        X, y, fh_ = self._format_dates(dates, X, y, predict=True)
-        if X is not None and not X.empty:
-            X = X.select_dtypes(exclude=["datetime64"])
+        fh_ = self._set_forecast(X)
+        X = X[self.cols_to_keep]
+        X = X.select_dtypes(exclude=["datetime64"])
```

Comment:
`_remove_datetime` isn't called here because ARIMA doesn't have a problem predicting on data that has an arbitrary temporal index, even if it was fit on data that doesn't have a temporal index (because the forecast horizon is relative).